Sources/Citadel/SFTP/SFTPServerInboundHandler.swift

```diff
@@ -0,0 +1,173 @@
+                    flags: command.pFlags
+                )
+            }
+            
+            promise.futureResult.map { file in
+                let handle = self.currentHandleID
+                self.files[handle] = file
+                self.currentHandleID &+= 1
+                
+                return SFTPMessage.handle(
+                    SFTPMessage.Handle(
+                        requestId: command.requestId,
+                        handle: ByteBuffer(integer: handle, endianness: .big)
+                    )
+                )
+            }.whenSuccess { handle in
+                context.writeAndFlush(NIOAny(handle), promise: nil)
+            }
+        case .closeFile(let command):
+            var handle = command.handle
+            
+            guard
+                let id: UInt32 = handle.readInteger(),
+                handle.readableBytes == 0,
+                let file = files[id]
+            else {
+                logger.error("bad SFTP file handle")
+                return
+            }
+            
+            let promise = context.eventLoop.makePromise(of: SFTPStatusCode.self)
+            file.close(promise: promise)
+            files[id] = nil
+            promise.futureResult.flatMap { status in
+                context.channel.writeAndFlush(
+                    SFTPMessage.status(
+                        SFTPMessage.Status(
+                            requestId: command.requestId,
+                            errorCode: status,
+                            message: "uploaded",
+                            languageTag: "EN"
+                        )
+                    )
+                )
+            }.whenFailure { _ in
+                context.channel.close(promise: nil)
+            }
+        case .read(let command):
+            print(command)
+        case .write(let command):
+            var handle = command.handle
+            
+            guard
+                let id: UInt32 = handle.readInteger(),
+                handle.readableBytes == 0,
+                let file = files[id]
+            else {
+                logger.error("bad SFTP file andle")
+                return
+            }
+            
+            let promise = context.eventLoop.makePromise(of: SFTPStatusCode.self)
+            file.write(command.data, atOffset: command.offset, promise: promise)
+            promise.futureResult.flatMap { status in
+                context.channel.writeAndFlush(
+                    SFTPMessage.status(
+                        SFTPMessage.Status(
+                            requestId: command.requestId,
+                            errorCode: status,
+                            message: "",
+                            languageTag: "EN"
+                        )
+                    )
+                )
+            }.whenFailure { _ in
+                context.channel.close(promise: nil)
+            }
+        case .handle(let command):
+            print(command)
```

Comment:
Are these print statements needed? Not sure what their purpose is, or maybe replace by a log.trace? Let me know.

Reply:
They still have to be forwarded to SFTPDelegate

Reply:
`handle` should throw an error, and `status` and `data`, too. `mkdir` should still be forwarded to the delegate